
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Tblastn protein alignments with refinement - dictyBase Developers</title>
  <meta name="author" content="dictyBase Developers">

  
  <meta name="description" content="Tblastn Protein Alignments With Refinement Mar 20th, 2013 Tblastn alignments needs to be refined as they do not properly align around exon intron &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://dictyBase.github.com/refining-tblastn-protein-alignments">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="dictyBase Developers" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="./javascripts/lib/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
<script type="text/javascript">
  // Revert jQuery's '$' alias,  to avoid clashes with ender.js. NOTE: Use
    // jQuery(...),  instead of $(...) from here on.
      jQuery.noConflict();
</script>
<script src="/javascripts/jquery.tableofcontents.min.js" type="text/javascript"></script>
<script src="/javascripts/generate-toc.js" type="text/javascript"></script>


  

</head>

<body   >
  <header role="banner"><hgroup>
  <h1><a href="/">dictyBase Developers</a></h1>
  
    <h2>Solving one problem at a time</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:dictyBase.github.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/pages.html">Pages</a></li>
  <li><a href="/pages/documentation">Documentation</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article role="article">
  
  <header>
    <h1 class="entry-title">Tblastn Protein Alignments With Refinement</h1>
    <p class="meta">








  


<time datetime="2013-03-20T16:40:00-05:00" pubdate data-updated="true">Mar 20<span>th</span>, 2013</time></p>
  </header>
  
  <p>Tblastn alignments needs to be refined as they do not properly align around exon intron
boundaries,  group hsps wide apart,  also might get aligned to the repeat regions. The
approach would be &hellip;</p>

<ul>
<li>Align against a repeat masked database.</li>
<li>Repeat filtering with dust</li>
<li>Run tblastn and specify <code>max-intron-lengh</code>.</li>
<li>Further post process the alignments and convert to gff3.</li>
</ul>


<h3>Mask genome sequence</h3>

<p>   An example with pallidum supercontig</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>    dustmasker -in pallidum_supercontig.fasta -infmt fasta <span class="se">\</span>
</span><span class='line'>       -parse_seqids -outfmt maskinfo_asn1_bin  <span class="se">\</span>
</span><span class='line'>       -out pallidum_supercontig_dust.asnb
</span></code></pre></td></tr></table></div></figure>


<h3>Make blast database with masking information</h3>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>    makeblastdb -in pallidum_supercontig.fasta -dbtype nucl -parse_seqids <span class="se">\</span>
</span><span class='line'>        -mask_data pallidum_supercontig_dust.asnb -out pallidum_supercontig
</span></code></pre></td></tr></table></div></figure>


<h3>Run tblastn</h3>

<p>It&rsquo;s a mapping protocol, so the search is expected to be on the insensitive side. The parameters are going to be selected to make the search as stringent as possible without loosing the top alignments.</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>     tblastn -evalue 1e-10 -threshold 999 -max_intron_length 3000  <span class="se">\</span>
</span><span class='line'>         -outfmt 5 -db_soft_mask 11 -max_target_seqs 5 <span class="se">\</span>
</span><span class='line'>         -num_threads 8  -db pallidum_supercontig
</span><span class='line'>         -query discoideum_polypeptide.fasta  <span class="se">\</span>
</span><span class='line'>         -out pallidum/discoideum_tblastn.xml
</span></code></pre></td></tr></table></div></figure>


<ul>
<li><strong>e-value          :</strong> 1e-10, we have tried this before and seems a good starting point.</li>
<li><strong>threshold        :</strong>  999, Neighborhood threshold value, a high value makes the seeding phase extremely stringent.</li>
<li><strong>db_soft_mask     :</strong>  Uses soft masking in database generated before by dustmasker.</li>
<li><strong>max_target_seqs  :</strong>  Maximum no of hits per discoideum protein that will be reported in the output. In this case it will only have top five alignments.</li>
</ul>


<h3>Refine TBLASTN alignments and convert it to GFF3</h3>

<p> Run the  <em>blast2gbrowsegff3</em>  perl script from <a href="https://github.com/dictyBase/Modware-Loader/blob/develop/lib/Modware/Transform/Command/blast2gbrowsegff3.pm">Modware Loader</a> distribution.</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>perl -Ilib bin/modware-transform blast2gbrowsegff3 <span class="se">\</span>
</span><span class='line'>      -i pallidum/discoideum_tblastn.xml <span class="se">\</span>
</span><span class='line'>      -o pallidum/discoideum_tblastn.gff3 --format blastxml <span class="se">\</span>
</span><span class='line'>      --query_id_parser general  --desc_parser ncbi <span class="se">\</span>
</span><span class='line'>      --source tblastn.dictyBase  --max_intron_length 3000 <span class="se">\</span>
</span><span class='line'>      --merge_contained --remove_stop_codon
</span></code></pre></td></tr></table></div></figure>


<p>Because of the nature of the blast statistics, blast aligns wherever it can. As a result, it has multiple alignments in the same region, sometimes the aligned regions are out of order and for translated blast it also extends through stop codons.
So, bunch of filtering options are used during the above conversion &hellip;</p>

<p><strong>max_intron_length:</strong> 3000 bp. It is the maximum distance(in bp) allowed between two consecutive hsps(High scoring pair) of a tblastn hit. Since, blast statistics do not account for splice junctions or exon intron models, it tends to group hsps that are far apart in the genome.  This option split hsps(into separate hit groups) that are more than <em>max_intron_length</em> apart. The maximum intron length (<code>3000 bp</code>)  is calculated from the multi exonic gene models(curated and predicted) of various dictyostelids(discoideum,purpureum,fasciculatum and pallidum).</p>

<p><strong>merge_contained:</strong> This activates merging of overlapping hsps(<code>high scoring pair</code>). In this case, the hsps whose start and end falls within another are merged with the larger one. The alignment attributes of larger ones are kept.</p>

<p><strong>remove_stop_codon:</strong> In the first step the hsps are regrouped based on their strand and frame of their alignment which means a single hit could be split into maximum possible of  <code>6 groups</code>.  In the next step, the genome(<code>subject</code>) sequence in hsp alignment is checked for extension through stop codon. If found, that hit along with the hsps are discarded.</p>

<p>Overall, the filtering options  helps to provide a best possible  contiguous alignment with one or more segments(<code>hsp</code>). The options are tuned for hsps aligned in the exon regions where two or more grouped hsps will flank the intervening introns. Few examples,</p>

<p><strong>Alignment with pallidum</strong></p>

<p><img src="http://genomes.dictybase.org/browser/gbrowse_img/pallidum/?name=PPA1366818:64600..66300;l=Gene%1EGenes%20Overview:region%1EGenemodel%1Ediscoideum_tblastn_alignments%1EContigs%20Overview:overview;width=1024;id=e366cd5d931987f0c2eae09e45696c42;format=GD;keystyle=between;grid=on;h_feat=eria@yellow" alt="pallidum alignment" /></p>

<p><strong>Alignment with purpureum</strong></p>

<p><img src="http://genomes.dictybase.org/browser/gbrowse_img/purpureum/?name=scaffold_112:1..20315;l=Gene%1EGenes%20Overview:overview%1EGenemodel%1EPredictions%1Ediscoideum_tblastn_alignments;width=800;id=e366cd5d931987f0c2eae09e45696c42;format=GD;keystyle=between;grid=1;h_feat=dpu_g0052220@yellow" alt="purpureum alignment" /></p>

  
    <footer>
      <p class="meta">
        
  

<span class="byline author vcard">Posted by <span class="fn">Siddhartha Basu</span></span>

        








  


<time datetime="2013-03-20T16:40:00-05:00" pubdate data-updated="true">Mar 20<span>th</span>, 2013</time>
        
      </p>
      
        <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://dictyBase.github.com/refining-tblastn-protein-alignments/index.html" data-via="dictybase" data-counturl="http://dictyBase.github.com/refining-tblastn-protein-alignments/index.html" >Tweet</a>
  
  
  
</div>

      
    </footer>
  
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>About </h1>
  <p>Unofficial blog of dictybase MOD(model organism database) developers.
  		<br/><a href="/about">More ....</a>
  </p>
</section>

<section>
  <h1>Latest Tweets</h1>
  <ul id="tweets">
    <li class="loading">Status updating...</li>
  </ul>
  <script type="text/javascript">
    $.domReady(function(){
      getTwitterFeed("dictybase", 4, false);
    });
  </script>
  <script src="/javascripts/twitter.js" type="text/javascript"> </script>
  
    <a href="http://twitter.com/dictybase" class="twitter-follow-button" data-show-count="false">Follow @dictybase</a>
  
</section>


<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/dictybase">@dictybase</a> on GitHub
  
  <script type="text/javascript">
    $(document).ready(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'dictybase',
            count: 5,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>

<section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2013/09/18/chado-loader-design/">Design Pattern of Chado Database Loaders</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/03/06/exporting-discoideum-annotations/">Exporting D.discoideum Annotations in GFF3 Format</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/28/taming-the-gff3/">Taming the Dictybase GFF3</a>
      </li>
    
  </ul>
</section>

  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - dictyBase Developers -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>







</body>
</html>
