
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Prepare and loading data for visualizing in gbrowse - dictyBase Developers</title>
  <meta name="author" content="dictyBase Developers">

  
  <meta name="description" content="Prepare and Loading Data for Visualizing in Gbrowse Mar 20th, 2013 The scripts in Modware-Loader distribution will be needed for data export and &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://dictyBase.github.com/prepare-data-for-gbrowse-visualization">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="dictyBase Developers" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="./javascripts/lib/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
<script type="text/javascript">
  // Revert jQuery's '$' alias,  to avoid clashes with ender.js. NOTE: Use
    // jQuery(...),  instead of $(...) from here on.
      jQuery.noConflict();
</script>
<script src="/javascripts/jquery.tableofcontents.min.js" type="text/javascript"></script>
<script src="/javascripts/generate-toc.js" type="text/javascript"></script>


  

</head>

<body   >
  <header role="banner"><hgroup>
  <h1><a href="/">dictyBase Developers</a></h1>
  
    <h2>Solving one problem at a time</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:dictyBase.github.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/pages.html">Pages</a></li>
  <li><a href="/pages/documentation">Documentation</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article role="article">
  
  <header>
    <h1 class="entry-title">Prepare and Loading Data for Visualizing in Gbrowse</h1>
    <p class="meta">








  


<time datetime="2013-03-20T16:12:00-05:00" pubdate data-updated="true">Mar 20<span>th</span>, 2013</time></p>
  </header>
  
  <p>The scripts in Modware-Loader distribution will be needed for data export and format conversions.</p>

<ul>
<li>First <a href="/install-modware-loader">install</a> Modware-Loader.</li>
</ul>


<h3>Export genome annotations in GFF3 format</h3>

<p>The annotations are exported from chado oracle database.</p>

<ul>
<li>Exporting purpureum genome</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$_</span>&gt; modware-export chado2gff3 --org purpureum --dsn <span class="s1">&#39;dbi:Oracle:host=host;sid=sid&#39;</span> <span class="se">\ </span>
</span><span class='line'>       -u chado -p chado  -o purpureum.gff3 --extra_gene_model Geneid <span class="se">\ </span>
</span><span class='line'>        --include_aligned_feature EST --tolerate_missing --include_align_parts
</span></code></pre></td></tr></table></div></figure>


<p>The above line exports the <em>Dictyostelium purpureum</em> genome in GFF3 format along with extra <strong>gene models</strong> and <strong>EST</strong> alignment features.
Many more examples of GFF3 exports are
<a href="/blog/2013/03/06/exporting-discoideum-annotations">here</a></p>

<h3>Including gene products in fasta header</h3>

<ul>
<li>Download the dictyostelium protein fasta <a href="http://dictybase.org/db/cgi-bin/dictyBase/download/download.pl?area=blast_databases&amp;ID=purpureum_protein.fas.gz">file</a></li>
<li><a href="https://gist.github.com/2023747">mapgeneid2prod.pl</a> : Produces a map between Gene ID and product name.</li>
<li><a href="https://gist.github.com/2023749">rewrite_dicty_fasta_header.pl</a> : Rewrite the fasta write to include the product name.</li>
</ul>


<h3>Aligning proteins to other genomes  using tblastn</h3>

<p>Here, for examples,  <em>Dictyostelium discoideum</em> proteins will be aligned to the top level assemblies(supercontigs) of pallidum genome. Detail alignment strategy with refinement is given <a href="/refining-tblastn-protein-alignments">here</a>.</p>

<h3>Loading features from GFF3</h3>

<p>The GFF3 data are going to loaded in Postgresql database using <a href="https://metacpan.org/module/Bio::DB::SeqFeature::Store">BioPerl&rsquo;s SeqFeature</a> backend.</p>

<ul>
<li>Install DBI backend for Postgresql</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>cpanm DBD::Pg
</span></code></pre></td></tr></table></div></figure>


<p><strong>Note:</strong> Do not set -Darch option while compiling perl with perlbrew, otherwise DBI won&rsquo;t get installed</p>

<ul>
<li>Load data</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>bp_seqfeature_load.pl --dsn <span class="s1">&#39;dbi:Pg:database=purpureum&#39;</span> -a <span class="s1">&#39;DBI::Pg&#39;</span> -u uuser <span class="se">\</span>
</span><span class='line'>-p passs -f --summary -c purpureum.gff3
</span></code></pre></td></tr></table></div></figure>


<p>If you load more feature in the same database just skip the -c options. However,
if it also complains about feature yet not found in the database then also skip the (-f) option.</p>

<p><strong>Note:</strong> For loading tblastn alignment skip the <strong>-c</strong>, but use the <strong>-f</strong> option.</p>

<ul>
<li>Edit and add database source in gbrowse configuration file.</li>
</ul>


<h3>RNA-seq(NGS) alignments data</h3>

<p>The alignments are expected to be in BAM format, if not run any standard NGS alignment pipeline(bowtie etc..) to get the BAM format.</p>

<ul>
<li>Create a folder under $GBROWSE_ROOT/database</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>mkdir -p  <span class="nv">$GBROWSE_ROOT</span>/database/purpureum
</span></code></pre></td></tr></table></div></figure>


<ul>
<li>Index the BAM file</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>samtools index file.bam
</span></code></pre></td></tr></table></div></figure>


<ul>
<li><p>Also copy the fasta sequence of reference genome in the same folder</p></li>
<li><p>Install perl binding for samtools</p></li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>cpanm Bio::DB::Sam
</span></code></pre></td></tr></table></div></figure>


<ul>
<li><strong>Note:</strong> If Bio::DB::Sam install fails, try to do it from source</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>  First download<span class="o">(</span>samtools.sf.net<span class="o">)</span> and compile samtools from <span class="nb">source</span>
</span><span class='line'><span class="nb">cd </span>samtools-&lt;version&gt;/
</span><span class='line'>make
</span><span class='line'>  It is fine <span class="k">if </span>it cannot compile tview, it can happen in absense of curses library. It is not needed <span class="k">for </span>the perl module
</span><span class='line'>
</span><span class='line'>  Now download Bio::DB::Sam <span class="nb">source</span> <span class="o">(</span>http://metacpan.org/module/Bio::DB::Sam<span class="o">)</span>
</span><span class='line'><span class="nb">cd </span>Bio-DB-SAM-&lt;version&gt;
</span><span class='line'><span class="nv">SAMTOOLS</span><span class="o">=</span>&lt;samtools-path&gt; perl Build.PL
</span><span class='line'>./Build install
</span></code></pre></td></tr></table></div></figure>


<ul>
<li>Edit and add the track configuration as described in this <a href="http://gmod.org/wiki/GBrowse_Volvox_SAM_Tutorial">guide</a></li>
</ul>


<h3><strong>Next:</strong> <a href="https://github.com/dictyBase/GBrowse-PSGI/wiki/Integrating-dictyBase-header-and-footer-gbrowse2">Integrate</a> headers and footers with gbrowse</h3>

  
    <footer>
      <p class="meta">
        
  

<span class="byline author vcard">Posted by <span class="fn">Siddhartha Basu</span></span>

        








  


<time datetime="2013-03-20T16:12:00-05:00" pubdate data-updated="true">Mar 20<span>th</span>, 2013</time>
        
      </p>
      
        <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://dictyBase.github.com/prepare-data-for-gbrowse-visualization/index.html" data-via="dictybase" data-counturl="http://dictyBase.github.com/prepare-data-for-gbrowse-visualization/index.html" >Tweet</a>
  
  
  
</div>

      
    </footer>
  
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>About </h1>
  <p>Unofficial blog of dictybase MOD(model organism database) developers.
  		<br/><a href="/about">More ....</a>
  </p>
</section>

<section>
  <h1>Latest Tweets</h1>
  <ul id="tweets">
    <li class="loading">Status updating...</li>
  </ul>
  <script type="text/javascript">
    $.domReady(function(){
      getTwitterFeed("dictybase", 4, false);
    });
  </script>
  <script src="/javascripts/twitter.js" type="text/javascript"> </script>
  
    <a href="http://twitter.com/dictybase" class="twitter-follow-button" data-show-count="false">Follow @dictybase</a>
  
</section>


<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/dictybase">@dictybase</a> on GitHub
  
  <script type="text/javascript">
    $(document).ready(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'dictybase',
            count: 5,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>

<section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2013/09/18/chado-loader-design/">Design Pattern of Chado Database Loaders</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/03/06/exporting-discoideum-annotations/">Exporting D.discoideum Annotations in GFF3 Format</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/28/taming-the-gff3/">Taming the Dictybase GFF3</a>
      </li>
    
  </ul>
</section>

  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2013 - dictyBase Developers -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>







</body>
</html>
