
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Installing Modware Loader - dictyBase Developers</title>
  <meta name="author" content="dictyBase Developers">

  
  <meta name="description" content="Installing Modware Loader Oct 29th, 2012 Modware-Loader distribution provides
command line scripts to import and export data from
chado relational &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://dictyBase.github.com/install-modware-loader">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="dictyBase Developers" type="application/atom+xml">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
  <script>!window.jQuery && document.write(unescape('%3Cscript src="./javascripts/lib/jquery.min.js"%3E%3C/script%3E'))</script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
<script type="text/javascript">
  // Revert jQuery's '$' alias,  to avoid clashes with ender.js. NOTE: Use
    // jQuery(...),  instead of $(...) from here on.
      jQuery.noConflict();
</script>
<script src="/javascripts/jquery.tableofcontents.min.js" type="text/javascript"></script>
<script src="/javascripts/generate-toc.js" type="text/javascript"></script>


  

</head>

<body   >
  <header role="banner"><hgroup>
  <h1><a href="/">dictyBase Developers</a></h1>
  
    <h2>Solving one problem at a time</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:dictyBase.github.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/pages.html">Pages</a></li>
  <li><a href="/pages/documentation">Documentation</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article role="article">
  
  <header>
    <h1 class="entry-title">Installing Modware Loader</h1>
    <p class="meta">








  


<time datetime="2012-10-29T14:53:00-05:00" pubdate data-updated="true">Oct 29<span>th</span>, 2012</time></p>
  </header>
  
  <p><a href="https://github.com/dictyBase/Modware-Loader">Modware-Loader</a> distribution provides
command line scripts to import and export data from
<a href="http://gmod.org/wiki/Introduction_to_Chado">chado</a> relational database in various
formats.</p>

<ul>
<li>Clone the Modware-Loader <a href="https://github.com/dictyBase/Modware-Loader">repository</a> and
switch to develop branch</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$_</span>&gt; git clone git://github.com/dictyBase/Modware-Loader.git
</span></code></pre></td></tr></table></div></figure>


<ul>
<li>Install the dependencies</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$_</span>&gt; <span class="nb">cd </span>Modware-Loader
</span><span class='line'><span class="nv">$_</span>&gt; cpanm -n Dist::Zilla
</span><span class='line'><span class="nv">$_</span>&gt; dzil authordeps | cpanm -n
</span><span class='line'><span class="nv">$_</span>&gt; dzil listdeps | cpanm -n
</span></code></pre></td></tr></table></div></figure>


<ul>
<li>Run any of <strong>modware</strong> command to list the sub-commands.</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$_</span>&gt; perl -Ilib bin/modware-export commands
</span><span class='line'>
</span><span class='line'>modware-export &lt;<span class="nb">command</span>&gt;
</span><span class='line'>
</span><span class='line'>Available commands:
</span><span class='line'>
</span><span class='line'>                       commands: list the application<span class="s1">&#39;s commands</span>
</span><span class='line'><span class="s1">                           help: display a command&#39;</span>s <span class="nb">help </span>screen
</span><span class='line'>
</span><span class='line'>            chado2alignmentgff3:  Export alignment from chado database in GFF3 format
</span><span class='line'>            chado2canonicalgff3:  Export canonical gene models from chado database in GFF3 format
</span><span class='line'>       chado2dictycanonicalgff3: Export GFF3 with canonical gene models of Dictyostelium discoideum
</span><span class='line'>         chado2dictycuratedgff3: Export GFF3 with curated gene models of Dictyostelium discoideum
</span><span class='line'>    chado2dictynoncanonicalgff3: Export GFF3 with sequencing center gene models of Dictyostelium discoideum
</span><span class='line'>  chado2dictynoncanonicalv2gff3:  Export GFF3 with repredicted gene models of Dictyostelium discoideum
</span><span class='line'>       chado2dictynoncodinggff3: Export GFF3 with non coding gene models of Dictyostelium discoideum
</span><span class='line'>                    chado2fasta: Export fasta sequence file from chado database
</span></code></pre></td></tr></table></div></figure>


<ul>
<li>Run any of the subcommands to display its options</li>
</ul>


<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$_</span>&gt; perl -Ilib bin/modware-export chado2canonicalgff3 <span class="nb">help</span>
</span><span class='line'>
</span><span class='line'>modware-export chado2canonicalgff3 <span class="o">[</span>-?chlopu<span class="o">]</span> <span class="o">[</span>long options...<span class="o">]</span>
</span><span class='line'>  --write_sequence            To write the fasta sequence<span class="o">(</span>s<span class="o">)</span> of
</span><span class='line'>                              reference feature<span class="o">(</span>s<span class="o">)</span>,  default is <span class="nb">true</span>
</span><span class='line'>  -h -? --usage --help        Prints this usage information.
</span><span class='line'>  --exclude_mitochondrial     Exclude mitochondrial genome,  default is
</span><span class='line'>                              to include <span class="k">if </span>it is present
</span><span class='line'>  --only_mitochondrial        Output only mitochondrial genome <span class="k">if </span>it is
</span><span class='line'>                              present,  default is <span class="nb">false</span>
</span><span class='line'>  -o --output                 Name of the output file,  <span class="k">if </span>absent
</span><span class='line'>                              writes to STDOUT
</span><span class='line'>  --feature_name              Output feature name instead of sequence
</span><span class='line'>                              id in the seq_id field,  default is off.
</span><span class='line'>  --attr --attribute          Additional database attribute
</span><span class='line'>  --rt --reference_type       The SO <span class="nb">type </span>of reference feature,
</span><span class='line'>                              default is supercontig
</span><span class='line'>  --reference_id              reference feature name/ID/accession
</span><span class='line'>                              number. In this <span class="k">case</span>,  only all of its
</span><span class='line'>                              associated features will be dumped. Takes
</span><span class='line'>                              precedence over dumping with
</span><span class='line'>                              mitochondrial options
</span><span class='line'>  --pass -p --password        database password
</span><span class='line'>  --dsn                       database DSN
</span><span class='line'>  --schema_debug              Output SQL statements that are executed,
</span><span class='line'>                              default to <span class="nb">false</span>
</span><span class='line'>  -u --user                   database user
</span><span class='line'>  --log_level                 Log level of the logger,  default is error
</span><span class='line'>  -l --logfile                Name of logfile,  default goes to STDERR
</span><span class='line'>  --species                   Name of species
</span><span class='line'>  --genus                     Name of the genus
</span><span class='line'>  -c --configfile             yaml config file to specify all <span class="nb">command</span>
</span><span class='line'><span class="nb">                             </span>line options
</span><span class='line'>  --org --organism            Common name of the organism whose genomic
</span><span class='line'>                              features will be exported
</span></code></pre></td></tr></table></div></figure>


<p>More or less any subcommands containing the word dicty is specific to work with
<a href="http://dictybase.org">D.discoideum</a> chado oracle database.</p>

  
    <footer>
      <p class="meta">
        
  

<span class="byline author vcard">Posted by <span class="fn">Siddhartha Basu</span></span>

        








  


<time datetime="2012-10-29T14:53:00-05:00" pubdate data-updated="true">Oct 29<span>th</span>, 2012</time>
        
      </p>
      
        <div class="sharing">
  
  <a href="//twitter.com/share" class="twitter-share-button" data-url="http://dictyBase.github.com/install-modware-loader/index.html" data-via="dictybase" data-counturl="http://dictyBase.github.com/install-modware-loader/index.html" >Tweet</a>
  
  
  
</div>

      
    </footer>
  
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>About </h1>
  <p>Unofficial blog of dictybase MOD(model organism database) developers.
  		<br/><a href="/about">More ....</a>
  </p>
</section>

<section>
  <h1>Latest Tweets</h1>
  <ul id="tweets">
    <li class="loading">Status updating...</li>
  </ul>
  <script type="text/javascript">
    $.domReady(function(){
      getTwitterFeed("dictybase", 4, false);
    });
  </script>
  <script src="/javascripts/twitter.js" type="text/javascript"> </script>
  
    <a href="http://twitter.com/dictybase" class="twitter-follow-button" data-show-count="false">Follow @dictybase</a>
  
</section>


<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/dictybase">@dictybase</a> on GitHub
  
  <script type="text/javascript">
    $(document).ready(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'dictybase',
            count: 5,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>

<section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2013/09/18/chado-loader-design/">Design Pattern of Chado Database Loaders</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/03/06/exporting-discoideum-annotations/">Exporting D.discoideum Annotations in GFF3 Format</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/28/taming-the-gff3/">Taming the Dictybase GFF3</a>
      </li>
    
  </ul>
</section>

  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - dictyBase Developers -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = '//platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>







</body>
</html>
